### PR TITLE
Doctest tutorial files

### PR DESCRIPTION
**Pull request by ev-br**
Try to make sure that our ReST tutorial stays current and does not give misleading examples. To this end, extend refguide_check to also check the contents of the doc/source/tutorial folder. This is partly an experiment to see how much one needs to bend over backwards to allow pseudocode where desirable and that there is no drive towards doctestablity for the sake of it (https://github.com/scipy/scipy/issues/4889#issuecomment-103282654). So far it seems reasonable, only minor whitelisting seems enough.

Travis Ci check is going to fail as there are several issues left. 
Am posting this PR early to (i) gather opinions on the concept, and (ii) to see how much small details vary between my machine and Travis box.

### CODE REVIEW

**Comment by ev-br** [comment]
OK, this seems to work: all tests pass for me locally, the only last failure left on Travis is somewhere in the plumbing of matplotlib and TNH does not seem related in any obvious way. 

The vast majority of changes to the rst files are minor syntax errors. There are several places where I tweaked the tutorials in a slightly less trivial way, so some review would be great. Esp tutorials for spatial, stats, optimize, interpolate and sparse.csgraph could use  a look-over.

File: doc/source/tutorial/csgraph.rst [hunk 1]
```diff
@@ -198,11 +198,11 @@ hand.  We can find the connecting list in the same way as above::
...
     >>> i = i2[0]
     >>> while i != i1[0]:
-    >>>     path.append(word_list[i])
-    >>>     i = predecessors[i1[0], i]
+    ...     path.append(word_list[i])
+    ...     i = predecessors[i1[0], i]
     >>> path.append(word_list[i1[0]])
     >>> print path[::-1]
-    ['imp', 'amp', 'asp', 'ask', 'ark', 'are', 'aye', 'rye', 'roe', 'woe', 'woo', 'who', 'oho', 'ohm']
+    ['imp', 'amp', 'asp', 'ass', 'ads', 'add', 'aid', 'mid', 'mod', 'moo', 'too', 'tho', 'oho', 'ohm']
```
Comment (ev-br) [inline doc/source/tutorial/csgraph.rst]: This one is interesting: the change reflects what is actually being output on both my laptop and on TravisCI. OTOH, the original version looks perfectly legit. I've a feeling there is a point worth discussing here, but it evades me at the moment.

**Comment by rgommers** [comment]
@ev-br do you want to merge this without enabling the TravisCI check, or work around the weird MPL issue?

File: doc/source/tutorial/csgraph.rst (hunk 1 above)
Comment (rgommers) [inline doc/source/tutorial/csgraph.rst]: Simply two paths of the same length, and due to internal code changes after the tutorial was written the other path is now found? This looks harmless to me.

**Comment by pv** [comment]
Some further fixup in https://github.com/ev-br/scipy/pull/4

**Comment by ev-br** [comment]
And it did fix the matplotlib failure! 
https://travis-ci.org/scipy/scipy/jobs/91233454

**Comment by pv** [comment]
LGMT, merged

**Pull request merged**